extension_templates/forecasting.py

```diff
@@ -12,6 +12,8 @@
 - make a copy of the template in a suitable location, give it a descriptive name.
 - work through all the "todo" comments below
 - fill in code for mandatory methods, and optionally for optional methods
+- do not write to reserved variables: is_fitted, _is_fitted, _X, _y, cutoff, _fh,
+    _cutoff, _converter_store_y, forecasters_, _tags, _tags_dynamic
```

Comment:
NOOOOOOOOOO

you are right, I made the same mistake!

Reply:
well... we could also change the base class?

Reply:
I see you went ahead and changed it in #2771.  I actually wonder if there would be value in changing the base class though??  Just because it seems like we are inconsistent with whether the protected names start or end with an underscore.  If we wouldn't break anything user facing by changing these things it could be worth it.  Otherwise, implemented the tests could be another way to help stop anyone from making these errors

Reply:
>  I actually wonder if there would be value in changing the base class though??

That's what I meant with my second comment. It would just be more difficult and will need reviews. Changing the multiplexer is "just" a bugfix.

> Just because it seems like we are inconsistent with whether the protected names start or end with an underscore.

Yes, you are right. I was also unhappy when I compiled the list and saw the result...